fe/src/main/java/org/apache/doris/common/util/DynamicPartitionUtil.java

```diff
@@ -0,0 +1,255 @@
+
+    public static void checkTimeUnit(String timeUnit) throws DdlException {
+        if (Strings.isNullOrEmpty(timeUnit)
+                || !(timeUnit.equalsIgnoreCase(TimeUnit.DAY.toString())
+                || timeUnit.equalsIgnoreCase(TimeUnit.WEEK.toString())
+                || timeUnit.equalsIgnoreCase(TimeUnit.MONTH.toString()))) {
+            ErrorReport.reportDdlException(ErrorCode.ERROR_DYNAMIC_PARTITION_TIME_UNIT, timeUnit);
+        }
+    }
+
+    private static void checkPrefix(String prefix) throws DdlException {
+        try {
+            FeNameFormat.checkPartitionName(prefix);
+        } catch (AnalysisException e) {
+            ErrorReport.reportDdlException(ErrorCode.ERROR_DYNAMIC_PARTITION_PREFIX, prefix);
+        }
+    }
+
+    private static void checkEnd(String end) throws DdlException {
+        if (Strings.isNullOrEmpty(end)) {
+            ErrorReport.reportDdlException(ErrorCode.ERROR_DYNAMIC_PARTITION_END_EMPTY);
+        }
+        try {
+            if (Integer.parseInt(end) <= 0) {
+                ErrorReport.reportDdlException(ErrorCode.ERROR_DYNAMIC_PARTITION_END_ZERO, end);
+            }
+        } catch (NumberFormatException e) {
+            ErrorReport.reportDdlException(ErrorCode.ERROR_DYNAMIC_PARTITION_END_FORMAT, end);
+        }
+    }
+
+    private static void checkBuckets(String buckets) throws DdlException {
+        if (Strings.isNullOrEmpty(buckets)) {
+            ErrorReport.reportDdlException(ErrorCode.ERROR_DYNAMIC_PARTITION_BUCKETS_EMPTY);
+        }
+        try {
+            if (Integer.parseInt(buckets) <= 0) {
+                ErrorReport.reportDdlException(ErrorCode.ERROR_DYNAMIC_PARTITION_BUCKETS_ZERO, buckets);
+            }
+        } catch (NumberFormatException e) {
+            ErrorReport.reportDdlException(ErrorCode.ERROR_DYNAMIC_PARTITION_BUCKETS_FORMAT, buckets);
+        }
+    }
+
+    private static void checkEnable(Database db, OlapTable tbl, String enable) throws DdlException {
+        if (Strings.isNullOrEmpty(enable)
+                || (!Boolean.TRUE.toString().equalsIgnoreCase(enable) && !Boolean.FALSE.toString().equalsIgnoreCase(enable))) {
+            ErrorReport.reportDdlException(ErrorCode.ERROR_DYNAMIC_PARTITION_ENABLE, enable);
+        }
+    }
+
+    public static boolean checkDynamicPartitionPropertiesExist(Map<String, String> properties) {
+        if (properties == null) {
+            return false;
+        }
+        return properties.containsKey(DynamicPartitionProperty.TIME_UNIT) ||
+                properties.containsKey(DynamicPartitionProperty.END) ||
+                properties.containsKey(DynamicPartitionProperty.PREFIX) ||
+                properties.containsKey(DynamicPartitionProperty.BUCKETS) ||
+                properties.containsKey(DynamicPartitionProperty.ENABLE);
+    }
+
+    public static void checkAllDynamicPartitionProperties(Map<String, String> properties) throws DdlException{
```

Comment:
> `checkDynamicPartitionPropertiesExist`
> `checkAllDynamicPartitionProperties`
> `analyzeDynamicPartition`
> 
> These 3 methods confused me a lot.... the properties should be checked once for all, no checking it many times.

chekAllDynamicPartitionProperties is use to check when user create table in case they forget add dynamic partition properties.
analyzeDynamicPartition is compatible with the logic of createOlapTable, remove the dynamic element in properties.
checkDynamicPartitionPropertiesExist is used to check when user change dynamic properties.

Reply:
I have change the method `chekAllDynamicPartitionProperties` to `checkInputDynamicPartitionProperties` to make it clear.